lucene/core/src/java/org/apache/lucene/search/AcceptDocs.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.search;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.lucene.util.BitSet;
+import org.apache.lucene.util.BitSetIterator;
+import org.apache.lucene.util.Bits;
+import org.apache.lucene.util.FixedBitSet;
+
+/**
+ * Higher-level abstraction for document acceptance filtering. Can be consumed in either
+ * random-access (Bits) or sequential (DocIdSetIterator) pattern.
+ *
+ * @lucene.experimental
+ */
+public abstract class AcceptDocs {
+
+  /**
+   * Random access to the accepted documents.
+   *
+   * @return Bits instance for random access, or null if not available
```

Comment:
I think we should apply the usual contract that a `null` Bits instance means that all docs are accepted?

```suggestion
   * @return Bits instance for random access, or null if all documents are accepted
```